pkg/reconciler/v1alpha1/autoscaling/kpa/kpa.go

```diff
@@ -285,6 +290,61 @@ func (c *Reconciler) reconcile(ctx context.Context, pa *pav1alpha1.PodAutoscaler
 	return nil
 }
 
+// IsReplaced returns true if it is known that the revision has been replaced,
```

Comment:
Finding no route to this revision is insufficient, because it could be the only revision of the configuration, and it could recover after a number of reboots or a fix to its dependencies.  We need to find a replacement to be sure that the revision won't be used and scaling it down does not affect the user.